### PR TITLE
Fixed windows share cannot modify frozen string bug

### PR DESCRIPTION
**Pull request by sanga1794**
## Description
Compilation fails with error "can't modify frozen String". It shows error at path in the windows_share method.

## Related Issue
fixed: https://github.com/chef/chef/issues/9166

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: spec/unit/resource/windows_share_spec.rb [hunk 1]
```diff
@@ -42,5 +42,9 @@
     expect(resource.path).to eql("C:\\chef")
     resource.path("C:\\chef")
     expect(resource.path).to eql("C:\\chef")
+    resource.path("C:/chef".freeze)
```
Comment (tas50) [inline spec/unit/resource/windows_share_spec.rb]: I'm not sure if we need these?

Comment (sanga1794) [reply spec/unit/resource/windows_share_spec.rb]: Okk. will update it. Thanks @tas50

Comment (tas50) [reply spec/unit/resource/windows_share_spec.rb]: If you're trying to make sure the path was frozen then you can just freeze the first one in this spec right?

Comment (sanga1794) [reply spec/unit/resource/windows_share_spec.rb]: Ok, I'll update it, thanks @tas50

Comment (sanga1794) [reply spec/unit/resource/windows_share_spec.rb]: Hello @tas50, All tests other than kitchen are passed.

**Pull request merged**